crates/csharp/src/lib.rs

```diff
@@ -1985,17 +2000,74 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                 let op = &operands[0];
                 results.push(format!("({}){}", t, op));
 
-                uwriteln!(
-                    self.src,
-                    "Debug.Assert(Enum.IsDefined(typeof({}), {}));",
-                    t,
-                    op
-                );
+                // uwriteln!(
+                //    self.src,
+                //    "Debug.Assert(Enum.IsDefined(typeof({}), {}));",
+                //    t,
+                //    op
+                // );
+            }
+
+            Instruction::ListCanonLower { element, realloc } => {
+                let op = &operands[0];
+                let (_size, ty) = list_element_info(element);
+
+                match self.gen.direction {
+                    Direction::Import => {
+                        let buffer: String = self.locals.tmp("buffer");
+                        uwrite!(
+                            self.src,
+                            "
+                            void* {buffer} = stackalloc {ty}[({op}).Length];
+                            {op}.AsSpan<{ty}>().CopyTo(new Span<{ty}>({buffer}, {op}.Length));
+                            "
+                        );
+                        results.push(format!("(int){buffer}"));
+                        results.push(format!("({op}).Length"));
+                    }
+                    Direction::Export => {
+                        // TODO: is the Alloc nessary here?
+                        let address = self.locals.tmp("address");
+                        let buffer = self.locals.tmp("buffer");
+                        let gc_handle = self.locals.tmp("gcHandle");
+                        uwrite!(
+                            self.src,
+                            "
+                        byte[] {buffer} = new byte[{op}.Length];
+                        Buffer.BlockCopy({op}, 0, {buffer}, 0, {op}.Length);
+                        var {gc_handle} = GCHandle.Alloc({op}, GCHandleType.Pinned);
+                        var {address} = {gc_handle}.AddrOfPinnedObject();
+                        Marshal.Copy({buffer}, 0, (IntPtr){address}, {op}.Length);                        
```

Comment:
Why do we need 2 copies, the `BlockCopy` and the `Marshal.Copy` ?  Shouldn't we be pinning `buffer` and using its address for `address`, and delete the `Marshal.Copy` ?

Reply:
I thought the same thing, but when I tried to clean this up it wouldn't get the correct values.  I can try it again...

Reply:
Not sure what I did last time but I removed the `Marshal.Copy` and it is working still